### ISSUE TITLE
SickGear on Synology Diskstation becoming unresponsive once every few days, nothing in the logs

### ISSUE THREAD

**Issue by infogrind** [SickGear/SickGear#1394]
#### Branch version commit hash: 1a89784 (though the issue has occurred in past versions too)

Expectation | Result
------------ | -------------
SickGear is always responsive | Once every 2-3 days or so (haven't measured statistics), SickGear becomes unresponsive. Meaning after I enter the URL in Chrome, it just keeps spinning, nothing happens. The log doesn't show any activity, nor any specific error. The only way to unblock it is to run `synopkgctl` to stop and start the package.

**Steps to reproduce:**
*1)* Install SickGear as a Synology Package on Synology Diskstation DS1511+ with 1 GB RAM at DSM 6.2.4-25556 Update 6.
*2)* Add about 200 TV series, download episodes, so that most shows are up to date with downloaded episodes.
*3)* Open the SickGear URL on a remote machine in Chrome.

I'm conscious that this report alone doesn't contain enough info to find the root cause of the issue. Rather, I'd like to ask what additional information I could provide, as there's really nothing in the log, even if it's at debug level.

**Comment by JackDandy** [comment]
Yup, you're right, cannot really help on this, as just too little info. 
SG can typically run at 350MB so you should be fine.

**Comment by infogrind** [comment]
Thanks for the fast reply! Like I said, I get that this is not enough info. But the problem still exists for me and I’d like to solve it somehow. 

Can you tell me what additional information I could provide, for more insight about the issue?

**Comment by JackDandy** [comment]
Perhaps your issue is a lack of memory as 1gb is not a lot, we have an RPi using near to that as a benchmark. Have you checked your swap files/partition, the RPi has 1.5gb total due to swapfiles. You should find out how much memory is free and being consumed **_when_** SG is unresponsive for you. The mem usage on Linux is much higher because of Python, and I would recommend PY3 as it is faster, PY2 EOL was 2020.

Perhaps check the  log file by reading it from disk to see what the last actions were. Maybe something ran of out mem at that point.

SG has never been unresponsive on Win (aside from PEBKAC), but I hope the pointers help you to reach the issue.

**Comment by infogrind** [comment]
Thanks, good ideas. Your comment made me realize that I'm still running SG from an old Synology package that doesn't support Python3. I've now downloaded the sickgear-specific Synology package following the instructions in your Wiki and – with a few hiccups (see the issue I just opened for it) – I could get it to work using Python3.

I'll keep an eye on the memory usage, and see if unresponsiveness coincides with high memory usage.
